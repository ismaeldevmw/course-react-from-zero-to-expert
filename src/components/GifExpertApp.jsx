import React, { useState } from 'react'
import { AddCategory } from './AddCategory';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai', 'Dragón'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai', 'Dragón']);

    // const handleAdd = () => {
    //     // setCategories([...categories, 'HunterXHunter']);
    //     setCategories( cats => [...cats, 'HunterXHunter']);
    // }

    return (
        <>

            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr/>

            <ol>
                {
                    categories.map( (category,  i) => {
                        return <li key={ i }>{ category }</li>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp
