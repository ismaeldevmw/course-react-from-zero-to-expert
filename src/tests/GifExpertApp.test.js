import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Test into GifExpertApp component', () => {
    
    test('should show <GifExpertApp /> alright', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show a list of categories', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe( categories.length );

    });

});