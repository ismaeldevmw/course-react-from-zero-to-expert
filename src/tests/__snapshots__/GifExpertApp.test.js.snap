// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into GifExpertApp component should show <GifExpertApp /> alright 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test into GifExpertApp component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
